src/main/java/sirius/kernel/commons/Strings.java

```diff
@@ -381,15 +391,108 @@ public static String limit(@Nullable Object input, int length) {
+     *         charactersToPreserveAtTheEnd. If charactersToPreserveAtTheEnd is greater than 0, the truncated signal
+     *         will have an additional ellipsis at the end, and therefore 13 characters in total.
+     *     </li>
+     *     <li>
+     *         end - this is the last part of the original string and contains the characters which should be
+     *         preserved at the end of the input.
+     *     </li>
+     * </ul>
+     *
+     * @param input                        the object which string representation should be limited to the given length
+     * @param length                       the max. number of characters to return
+     * @param charactersToPreserveAtTheEnd the number of characters to preserve at the end of the string
+     * @return the truncated string with a signal in the middle and characters preserved at the end, or the full string
+     * if the <tt>input</tt> is shorter than the given length + charactersToPreserveAtTheEnd. However,
+     * if input is <tt>null</tt>, "" is returned.
+     */
+    public static String truncateMiddle(@Nullable Object input, int length, int charactersToPreserveAtTheEnd) {
+        return limit(input, length, true, true, charactersToPreserveAtTheEnd);
+    }
+
+    /**
+     * Limits the length of the given string to the given length.
+     *
+     * @param input                        the object which string representation should be limited to the given length
+     * @param length                       the max. number of characters to return
+     * @param showEllipsis                 whether to append three dots if <tt>input</tt> is longer than <tt>length</tt>
+     * @param showTruncated                whether to append a marker/signal if <tt>input</tt> is longer than <tt>length</tt>
+     * @param charactersToPreserveAtTheEnd the number of characters to preserve at the end of the string
+     * @return a part of the string representation of the given <tt>input</tt>. If input is shorter
+     * than <tt>length</tt>, the full value is returned. If input is <tt>null</tt>, "" is returned.
+     * If <tt>charactersToPreserveAtTheEnd</tt> is negativ it is omitted.
+     */
+    private static String limit(@Nullable Object input,
+                                int length,
+                                boolean showEllipsis,
+                                boolean showTruncated,
+                                int charactersToPreserveAtTheEnd) {
         if (isEmpty(input)) {
             return "";
         }
+
+        if (length < 0) {
+            length = 0;
+        }
+
+        if (charactersToPreserveAtTheEnd < 0) {
+            charactersToPreserveAtTheEnd = 0;
+        }
+
         String str = String.valueOf(input).trim();
-        if (str.length() > length) {
-            return str.substring(0, (showEllipsis ? length - 1 : length)) + (showEllipsis ? "…" : "");
-        } else {
+        if (str.length() <= length + charactersToPreserveAtTheEnd) {
```

Comment:
**Optional:** would be nice to improve the variable name `str` as this is as unhelpful as it gets

Reply:
Don't really like the included type in the new name, but its okay :) better than before